test/framework/src/main/java/org/opensearch/cluster/coordination/AbstractCoordinatorTestCase.java

```diff
@@ -1173,7 +1176,8 @@ protected Optional<DisruptableMockTransport> getDisruptableMockTransport(Transpo
                     getElectionStrategy(),
                     nodeHealthService,
                     persistedStateRegistry,
-                    remoteStoreNodeService
+                    remoteStoreNodeService,
+                    Mockito.mock(MetricsRegistry.class)
```

Comment:
Should we rather construct `InMemoryMetricsRegistry` in-order for us to verify the counters collected during failures.

Reply:
Can we add the `InMemoryMetricsRegistry` to the tests as an implementation of MetricsRegistry that uses Hashmap for collecting the counters ?